### ISSUE TITLE
add polyfill for classList in IE11

### ISSUE THREAD

**Issue by seemajune** [kaliber5/ember-bootstrap#814]
IE11 does not have full support for  `classList`  so when this function runs on certain elements (like svgs) it will throw an error. A polyfill would be super helpful to fix this. Thank you!
---

Output from `ember bootstrap:info`:

```
TypeError: Unable to get property 'contains' of undefined or null reference
  at _click (addon-tree-output/ember-bootstrap/components/base/bs-modal/dialog.js:177:1)
  at o._run (backburner.js:1046:1)
  at o._join (backburner.js:1029:1)
  at o.join (backburner.js:749:1)
  at join (@ember/runloop/index.js:165:1)
...
(1 additional frame(s) were not displayed)
```

**Comment by simonihmig** [comment]
Fwiw, I made a classList polyfill addon before, back when IE9 was still supported: https://github.com/kaliber5/ember-cli-classlist-polyfill

But I am not sure if we really need this in this case.

@seemajune can you please describe the situation when the error with the above stack trace happened? Was this an SVG inside a modal, that was being clicked on?

**Comment by seemajune** [comment]
Hi @simonihmig, thanks for your quick reply and assistance!

It's actually occurring when an svg is clicked on to launch an ember-bootstrap modal in IE11, here is a better screenshot:

<img width="858" alt="Screen Shot 2019-05-17 at 6 06 53 PM" src="https://user-images.githubusercontent.com/2113081/57958732-c8dc7d00-78ce-11e9-903c-ba65246f88aa.png">

so i think a fallback for the lack of IE11 support for `classList` on svg elements could be a regex on the  `.className` - just an idea

**Comment by simonihmig** [comment]
The use of `classList` was actually pretty stupid here, I found a simpler implementation that should fix this here: #815.

@seemajune could you try that branch (`kaliber5/ember-bootstrap#fix-classlist-ie11` in your package.json) to see if it fixes your problem?

**Comment by seemajune** [comment]
> The use of `classList` was actually pretty stupid here, I found a simpler implementation that should fix this here: #815.
> 
> @seemajune could you try that branch (`kaliber5/ember-bootstrap#fix-classlist-ie11` in your package.json) to see if it fixes your problem?

It works 🎉 
Thank you SO much for being quick to fix and reply. It's nice to have people like you in the community. <3

**Comment by simonihmig** [comment]
@seemajune thank you for your kind words, much appreciated!

The fix has been released as v2.7.1!

**Comment by seemajune** [comment]
> @seemajune thank you for your kind words, much appreciated!
> 
> The fix has been released as v2.7.1!

thank you ! 😺
